src/functionaltest/java/com/ericsson/ei/frontend/SubscriptionHandlingFunctionality.java

```diff
@@ -56,10 +56,8 @@ public void testSubscription() throws Exception {
         String editButtonXPath = "//tr[td[contains(.,'Subscription1')]]/td/button[contains(text(),'Edit')]";
         String viewButtonXPath = "//tr[td[contains(.,'Subscription1')]]/td/button[contains(text(),'View')]";
         subscriptionPage.clickReload(response);
-        assert (new WebDriverWait(driver, 10)
-                .until((webdriver) -> ((driver.getPageSource().contains("Subscription1")))));
-        assert (new WebDriverWait(driver, 10)
-                .until((webdriver) -> ((driver.getPageSource().contains("Subscription2")))));
+        assert (subscriptionPage.textExistsInTable("//tr[td[contains(.,'Subscription1')]]"));
```

Comment:
When using a "generic" interaction with the driver it removes the reason why we chose to go for pageobjects. If there is a change in the html code this would have to be changed in multiple places instead of just 1 place.

What you could do instead of creating basically the same method is:

subscriptionInTable("Subscription1");

And then in subscriptionInTable you just add the strings together so that its possible to choose which subscription but the xpath would be hidden atleast.